### PR TITLE
feat: Main Focus with all functionalities

### PR DESCRIPTION
**Pull request by maitrakhatri**
# features added
- Add main focus
- Edit main focus 
- Main focus gets stored in local storage, so doesn't get clear on refresh
- Yesterday's main focus automatically gets cleared when date is changed and asks for new day's main focus

# demo

https://user-images.githubusercontent.com/54719132/159422498-182dffd4-7837-4b1c-a179-d3ef5734fd7d.mp4

### CODE REVIEW

File: gozo-browser-extension/src/Components/MainFocus.jsx [hunk 1]
```diff
@@ -1,8 +1,67 @@
...
+    }
+
+    //Show Main Focus Component
+    function ShowMainFocus() {
+        return (
+            <div className="show-main-focus">
+                <h1>My main focus today is {displayFocus} </h1> <button onClick={editHandler}>Edit</button>
+            </div>
+        )
+    }
```
Comment (SanketDhabarde) [inline gozo-browser-extension/src/Components/MainFocus.jsx:22]: Why not move these 2 components to separate files.

**Review by SanketDhabarde — APPROVED** [review]
- Like how you are providing comments to code for understanding.
- You can create a hook for localStorage things with setItem and getItem
- You can move functions to utils and make it a little generic. Just suggestion

File: gozo-browser-extension/src/Components/MainFocus.jsx (hunk 1 above)
Comment (maitrakhatri) [reply gozo-browser-extension/src/Components/MainFocus.jsx:22]: because they have just 2 line of code 😅

Comment (maitrakhatri) [reply gozo-browser-extension/src/Components/MainFocus.jsx:22]: because they have just 2 line of code 😅

**Comment by maitrakhatri** [comment]
> * Like how you are providing comments to code for understanding.
> * You can create a hook for localStorage things with setItem and getItem
> * You can move functions to utils and make it a little generic. Just suggestion

hook one is really good idea, thanks !! I'll implement it
I tried abstracting functions but I was facing some issues so decided to do it at the end

**Review by anupkgurung — APPROVED** [review]
suggestion : try to use arrow function all over  as use state uses arrow. Else u can also change arrow to normal function expression within use state so that entire file has only one pattern.

**Comment by maitrakhatri** [comment]
> suggestion : try to use arrow function all over as use state uses arrow. Else u can also change arrow to normal function expression within use state so that entire file has only one pattern.

I'll take care of that

**Pull request merged**